autoscheduler/scraper/tests/api_tests.py

```diff
@@ -265,3 +271,48 @@ def test_api_course_search_gives_correct_results_c(self):
         # Assert
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response.json(), expected)
+
+    def test_api_course_search_gives_correct_results_csce_3(self):
+        """ Tests that /api/course/search?search=CSCE%203&term=201731 gives correct
+            output
+        """
+        # Arrange
+        expected = {'results': ['CSCE 310', 'CSCE 315']}
+        data = {'search': 'CSCE%203', 'term': '201731'}
+
+        # Act
+        response = self.client.get('/api/course/search', data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), expected)
+
+    def test_api_course_search_gives_correct_results_csce_lower(self):
+        """ Tests that /api/course/search?search=csce&term=201731 gives correct
+            output
+        """
+        # Arrange
+        expected = {'results': ['CSCE 181', 'CSCE 310', 'CSCE 315']}
+        data = {'search': 'csce', 'term': '201731'}
+
+        # Act
+        response = self.client.get('/api/course/search', data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), expected)
+
+    def test_api_course_search_gives_correct_results_csce_3_lower(self):
```

Comment:
This last test might not be strictly necessary, since each feature (lowercase and number) works individually and there's no reason for them to interfere with other. However, I wouldn't change it now that it's been done. This PR looks good to me, I'll just let one of the backend guys look at it once for final approval.